### PR TITLE
[SPARK-20672][SS] Keep the `isStreaming` property in triggerLogicalPlan in Structured Streaming

### PR DESCRIPTION
**Pull request by uncleGen**
## What changes were proposed in this pull request?

In Structured Streaming, the "isStreaming" property will be eliminated in each triggerLogicalPlan. Then, some rules will be applied to this triggerLogicalPlan mistakenly. So, we should refactor existing code to better execute batch query and ss query.

Like in #17896, we will eliminate `EventTimeWatermark` in any batch query. If we add this rule into `Analyzer.batches`, we will eliminate the `EventTimeWatermark` in triggerLogicalPlan mistakenly, as it is also a `batch query` in current implementation.

## How was this patch tested?

existing ut.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala [hunk 1]
```diff
@@ -64,8 +64,20 @@ case class StreamingRelationExec(sourceName: String, output: Seq[Attribute]) ext
...
 }
 
-object StreamingExecutionRelation {
-  def apply(source: Source): StreamingExecutionRelation = {
-    StreamingExecutionRelation(source, source.schema.toAttributes)
+case class StreamingRelationWrapper(child: LogicalPlan) extends UnaryNode {
+  override def isStreaming: Boolean = true
+  override def output: Seq[Attribute] = child.output
+}
+
```
Comment (uncleGen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala]: Add a new `StreamingRelationWrapper` relation to wrap the internal relation in each trigger. It keeps the `isStreaming` property.

**Comment by uncleGen** [comment]
retest this please.

**Comment by uncleGen** [comment]
cc @zsxwing

File: sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ case class StreamingRelation(dataSource: DataSource, sourceName: String, output:
  * Used to link a streaming [[Source]] of data into a
  * [[org.apache.spark.sql.catalyst.plans.logical.LogicalPlan]].
  */
-case class StreamingExecutionRelation(source: Source, output: Seq[Attribute]) extends LeafNode {
+case class StreamingSourceRelation(source: Source, output: Seq[Attribute]) extends LeafNode {
   override def isStreaming: Boolean = true
```
Comment (uncleGen) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingRelation.scala]: just one renaming

**Comment by zsxwing** [comment]
@uncleGen Thanks for doing this. As adding an intermediate node into the logical plan may make some optimizations fail, we need to find out a better solution. We probably will make a big refactoring in 2.3 to fix this issue from the root.

**Comment by uncleGen** [comment]
@zsxwing Great! Close this pr then.

**Pull request closed without merging**